src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/UpdateAzureContainerRegistry.cs

```diff
@@ -15,68 +15,60 @@
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Container Registry Name.")]
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = EnableAdminUserByResourceNameParameterSet, HelpMessage = "Container Registry Name.")]
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = DisableAdminUserByResourceNameParameterSet, HelpMessage = "Container Registry Name.")]
         [Alias(ContainerRegistryNameAlias, RegistryNameAlias, ResourceNameAlias)]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
-        [Parameter(
-            Mandatory = true,
-            ParameterSetName = EnableAdminUserParameterSet,
-            HelpMessage = "Enable admin user for the container registry.")]
-        [Parameter(
-            Mandatory = false,
-            ParameterSetName = DisableAdminUserParameterSet,
-            HelpMessage = "Disable admin user for the container registry.")]
+        [Parameter(Mandatory = true, ParameterSetName = EnableAdminUserByResourceNameParameterSet, HelpMessage = "Enable admin user for the container registry.")]
+        [Parameter(Mandatory = true, ParameterSetName = EnableAdminUserByResourceIdParameterSet, HelpMessage = "Enable admin user for the container registry.")]
         [ValidateNotNull]
         [Alias(EnableAdminAlias)]
         public SwitchParameter EnableAdminUser { get; set; }
 
-        [Parameter(
-            Mandatory = false,
-            ParameterSetName = EnableAdminUserParameterSet,
-            HelpMessage = "Enable admin user for the container registry.")]
-        [Parameter(
-            Mandatory = true,
-            ParameterSetName = DisableAdminUserParameterSet,
-            HelpMessage = "Disable admin user for the container registry.")]
+        [Parameter(Mandatory = true, ParameterSetName = DisableAdminUserByResourceNameParameterSet, HelpMessage = "Disable admin user for the container registry.")]
+        [Parameter(Mandatory = true, ParameterSetName = DisableAdminUserByResourceIdParameterSet, HelpMessage = "Disable admin user for the container registry.")]
         [ValidateNotNull]
         [Alias(DisableAdminAlias)]
         public SwitchParameter DisableAdminUser { get; set; }
 
-        [Parameter(
-            Mandatory = false,
-            HelpMessage = "Container Registry Tags.")]
+        [Parameter(Mandatory = false, HelpMessage = "Container Registry Tags.")]
         [ValidateNotNull]
         [Alias(TagsAlias)]
         public Hashtable Tag { get; set; }
 
-        [Parameter(
-            Mandatory = false,
-            HelpMessage = "The name of an existing storage account.")]
+        [Parameter(Mandatory = false, HelpMessage = "The name of an existing storage account.")]
+        [ValidateNotNullOrEmpty]
         public string StorageAccountName { get; set; }
 
+        [Parameter(Position = 2, Mandatory = false, HelpMessage = "Container Registry SKU.")]
```

Comment:
@Jinhuafei I would remove the `Position` from this parameter since it is optional

Reply:
OK.